server/src/main/java/org/apache/iotdb/db/engine/snapshot/SnapshotTaker.java

```diff
@@ -89,111 +70,131 @@ public boolean takeFullSnapshot(String snapshotDirPath, boolean flushBeforeSnaps
       dataRegion.syncCloseAllWorkingTsFileProcessors();
     }
 
-    List<Long> timePartitions = dataRegion.getTimePartitions();
+    File snapshotLog = new File(snapshotDir, SnapshotLogger.SNAPSHOT_LOG_NAME);
+    try {
+      snapshotLogger = new SnapshotLogger(snapshotLog);
+      boolean success = false;
+
+      readLockTheFile();
+      try {
+        success = createSnapshot(seqFiles, snapshotDir.getName()) || success;
+        success = createSnapshot(unseqFiles, snapshotDir.getName()) || success;
+      } finally {
+        readUnlockTheFile();
+      }
+
+      LOGGER.info(
+          "Successfully take snapshot for {}-{}, snapshot directory is {}",
```

Comment:
What if the createSnapshot fail?  You need to check the success then log

Reply:
resolve